i2mv/methods.go

```diff
@@ -0,0 +1,75 @@
+package i2mv
+
+import (
+	"fmt"
+	"io"
+
+	symbols "github.com/taubyte/go-sdk-symbols/i2mv"
+)
+
+func Open(id uint32) (MemoryView, error) {
+	var size uint32
+	if err := symbols.MemoryViewSize(id, &size); err != 0 {
+		return nil, fmt.Errorf("getting memory view size failed with: %s", err)
+	}
+	return &memoryView{id: id, size: size}, nil
+}
+
+func (m *memoryView) Id() uint32 {
+	return m.id
+}
+
+func (m *memoryView) OffSet() int64 {
+	return m.offset
+}
+
+func (m *memoryView) Size() uint32 {
+	return m.size
+}
+
+func (m *memoryView) Read(p []byte) (int, error) {
+	var n uint32
+	if p == nil || len(p) == 0 {
```

Comment:
[]byte{nil}  will have len == 0

Reply:
There was a weird use case aron showed me a while ago where bytes were not nil, but with length 0

Reply:
[]byte == nil, []byte{} == 0

Reply:
Then just check len == 0

Reply:
[]byte == nil []byte{} == 0